### ISSUE TITLE
Can i implement OPC UA applikation like Server and like Client?

### ISSUE THREAD

**Issue by EvgKlin** [open62541/open62541#4342]
## Description
Can i implement OPC UA applikation like Server and like Client? Because i see some examples im Internet and it should be possible (see pic). 

## Background Information / Reproduction Steps![Unbenannt](https://user-images.githubusercontent.com/64699335/115011314-ca45c400-9eae-11eb-9751-34cef305923a.png)

Used CMake options:

```bash
cmake -DUA_NAMESPACE_ZERO=<YOUR_OPTION> <ANY_OTHER_OPTIONS> ..
```

## Checklist

Please provide the following information:

 - [ ] open62541 Version (release number or git tag):
 - [ ] Other OPC UA SDKs used (client or server): 
 - [ ] Operating system:
 - [ ] Logs (with `UA_LOGLEVEL` set as low as necessary) attached
 - [ ] Wireshark network dump attached
 - [ ] Self-contained code example attached
 - [ ] Critical issue

**Comment by jackybek** [comment]
Yes .. you can create 1 separate thread in your main() program : 
1. The main program will launch UA server
2. The main program spawns a new thread and within this thread you can launch a UA client.

**Comment by EvgKlin** [comment]
@jackybek Thank you. Can i ask one question: I can add nodes to server from the client. But can i add nodes from server to client? Or is it realized only over subscribing and request/response

**Comment by jackybek** [comment]
A opcua client does not have any nodes.  Only a opcua server has the nodes. So you can either:
1. Within a server code, add nodes to the address space
2. 2. Within a client code, send a request to the server to add new nodes

**Comment by EvgKlin** [comment]
@jackybek sorry for interrupting, there is other issue(https://github.com/open62541/open62541/issues/4353#issue-862735830), but i get the same problem. if use amalgamation in Windows, i get errors. In Debian it works perfectly. Maybe do i something wrong

**Comment by jackybek** [comment]
Sorry .. I am using Linux version and has not played with the windows version of open62541.

**Comment by andreasebner** [comment]
The original question seems to be answered. Amalgamation build are tested as part of the CI, so they should work.
